### DISCUSSION TITLE
[Feature] Windows & MAC OS Client

### DISCUSSION THREAD

**Discussion by AndyXheli** [immich-app/immich#5049] [Feature Request]
### The feature

Would this be passible for the future ? For syncing purposes 

### Platform

- [x] MAC OS
- [x] Windows

**Comment by alextran1502** [comment]
You can already download the app to use on M1 Mac and above

**Reply by AndyXheli** [reply]
Ahh okay, I have a Mac with intel chip.

**Comment by DevNoam** [comment]
To be honset, I really like the web version, I think its a bit worthless to have app.
But I do think it will be a nice addon to add some syncing software that will create a shared windows folder (Like iCloud does) with shortcut to open immich on the web..

**Reply by AndyXheli** [reply]
Yup that exactly what I'm taking about just a desktop Clint for syncing

**Comment by pec0ra** [comment]
If you consider developing a desktop client, I recommend also including linux compatibility. Some people will request it anyway and it may be easier if the compatibility is built in from the start.

**Reply by DavidThePurple** [reply]
Yes please!

**Reply by TomasVojta** [reply]
Definitely! I am using both Windows and Linux and I absolutely hate any Web apps! 😄 Please make the desktop app!

**Comment by landrix** [comment]
What about a PWA?

**Comment by ebbyb** [comment]
I'll second the OS app request. 

I love the web interface and phone apps, but someone in my family, who used Picasa pretty exclusively until it was canceled, has a hard time learning new software. 

Even if it was super basic with a similar layout mimicking Picasa, I think that would help them out a ton.

As always, keep up the awesome work!

**Reply by alextran1502** [reply]
You can already download the app from Apple App Store if you have Mac M1 and above

**Reply by ebbyb** [reply]
Yup, that's pretty cool. 

However they do not. :)

**Reply by lindomar-marques55** [reply]
I still use Picasa on Windows even though it's no longer supported or doesn't connect to the web.

**Comment by panialex** [comment]
maby star with the bulk upload, wich i think is more important to start using it

**Reply by KatieTheDev** [reply]
Bulk upload can be done with Immich CLI.

**Comment by scoop96** [comment]
So far im mixing immich and nextcloud to keep both parts, (upload and so i will use the nextcloud app to keep it consistent) but as soon as immich windows client is enabled i will switch the way of working.

**Comment by bo0tzz** [comment]
#1427

**Comment by cgonIT** [comment]
Hi! See that there long time without news on this topic. 
Is it planned in a near future to implement Windows Desktop Immich Client?
Thank you so much in advanced for the effort you do.

**Reply by bo0tzz** [reply]
We have no plans to implement this.

**Reply by TomasVojta** [reply]
Please reconsider it, there are so many annoying downsides in using web-app!
Look through all these comments...

**Comment by amarendra** [comment]
Does it work on Mac as  a backup tool as well or as a sync/export kinda tool?

**Comment by TypicalZedF** [comment]
Why was this closed? I genuinely believe there's a strong need for a dedicated syncing client for Windows.

People keep suggesting Immich CLI, and while it’s a useful tool, it’s not what many of us are asking for. What we really need is a reliable way to automatically sync specific folders—like the image folder on my desktop—directly to Immich.

For example, I have apps that save game clips or screenshots from VR sessions with friends. I want those to automatically upload to Immich without manual intervention. This is something Google Drive handles seamlessly with Google Photos integration—why can't Immich offer a similar desktop client?

A lightweight sync client that watches specific folders and pushes new content to Immich would be a game-changer for users looking to fully replace services like Google Photos.

**Reply by TomasVojta** [reply]
Both for Windows and Linux! 🫶

Using web-apps is always pain - requires more steps to get things done, can't drag-n-drop, right-click menu does web stuff, things load slower, can't alt-tab...

**Comment by lorniany** [comment]
I've built a small desktop application for Windows that monitors folders and connected SD cards, and can automatically upload new files to the Immich cloud.  
I can also add support for other platforms if there's interest.

[https://immisync.astily.dev/](https://immisync.astily.dev/)

P.S. It might be useful for this discussion since people here are looking for such a client.

**Reply by leisegang** [reply]
> I've built a small desktop application for Windows that monitors folders and connected SD cards, and can automatically upload new files to the Immich cloud. I can also add support for other platforms if there's interest.
> 
> https://immisync.astily.dev/
> 
> P.S. It might be useful for this discussion since people here are looking for such a client.

It doesnt load here:
<img width="792" height="624" alt="image" src="https://github.com/user-attachments/assets/d05b72dd-48ba-47ea-acdc-e7950ba6c59b" />

**Reply by TypicalZedF** [reply]
Works for me. IDK how reliable this is though, the software they sent doesn't seem to be open source... I can't find the code for it and there is no reference to a github project on the website.

**Reply by lorniany** [reply]
> > I've built a small desktop application for Windows that monitors folders and connected SD cards, and can automatically upload new files to the Immich cloud. I can also add support for other platforms if there's interest.
> > 
> > https://immisync.astily.dev/
> > 
> > P.S. It might be useful for this discussion since people here are looking for such a client.
> 
> It doesnt load here:
> 

Fixed, the issue was caused by the light theme.
Just plug in the flash drive/SD card and it will appear in the interface, or click the + at the top to add a folder on your computer for uploading. (The "+" button was not visible due to a light theme bug.)

P.S. I’m planning to eventually release the code publicly on GitHub, but I need to prepare the repository first. The app was originally developed for personal use, so some commit history might contain credentials or similar sensitive data. For now, my focus is on another project that’s easier to monetize rather than relying on donations.

Regarding privacy, the only data I collect is basic info like a random installation ID and the app version—purely to gauge interest and decide whether further development is worth it. It also does not perform any actions beyond what is expected. Since the code isn’t publicly available, there’s no reason to take my word for it, but still.

**Reply by TypicalZedF** [reply]
Nice to hear the light theme bug got sorted, that kind of stuff makes a big difference.

That said, I’d still need to wait until the code’s up on GitHub before I’d feel comfortable trying it out. The site should really link to the repo too, both for releases and transparency. Until then, I don’t think it’s worth the risk.

Good luck getting the repo cleaned up though, I know that can be a headache. Hopefully it works out, because an open-source client like this would be super useful.

**Reply by pannal** [reply]
Is this dead? The link leads nowhere.

**Reply by lorniany** [reply]
Yes, the project is currently abandoned. The website is down and the app is no longer supported, but I’ve attached the last working build here. I may bring the website back and continue development someday, but for now that seems unlikely.

[https://drive.google.com/file/d/1AN07wgD7h4G2WRKvAW39MIX2F79qJlpg/view?usp=sharing](https://drive.google.com/file/d/1AN07wgD7h4G2WRKvAW39MIX2F79qJlpg/view?usp=sharing)

**Comment by kaysond** [comment]
I'd also like a Windows/Linux desktop. The web client is great, but if I want to use any of my photos in some way, it requires so many extra steps: open browser, bring up immich, log in, find photo, download, find download, maybe unzip, and so on. It's just clunky. Something like the Mac OS Photos app with a sync feature (or even virtual filesystem support) would make things so much easier.

**Comment by CloisteredCaiman** [comment]
Yes, would love a linux desktop app. A shame this feature was close, I guess its not a priority for dev?

**Comment by Toga2007** [comment]
Yes, an app like this is sorely needed. Maybe the developers will finally take a look at this thread.

**Comment by kaysond** [comment]
I wonder how hard it would be to wrap the existing web front end in something like https://github.com/tauri-apps/tauri

The sync engine already exists in the mobile app so that could possibly be added too (or rewrite it in rust 🤣).

Of course the devil's in the details, least of which is differences in valid filenames...

**Comment by TariqMK** [comment]
Adding my vote for a Windows/Linux Client. Would be brilliant.

**Comment by su2den** [comment]
Please don't underestimate the importance of 'native' apps on every platform — I don't care if they're actually Electron. If you want more users, you have to adapt to their habits. There also needs to be a sync feature on every desktop. The current methods for moving a large iCloud library are very inconvenient.

**Comment by nicx17** [comment]
I built [Mimick](https://github.com/nicx17/mimick)

**Reply by CloisteredCaiman** [reply]
I absolutely love this on my PostmarketOS phone

**Comment by TomasVojta** [comment]
This is one thing I am truly missing on immich!

I **HATE** web apps. They are slower, have to occupy tabs, are limited to web environment, can't alt-tab to them, often don't support right-click or drag and drop, can't access physical data... Web app also makes you do many extra steps when you want to get the image in/out: hit download > find the file > move that file... Desktop app would just allow you to drag and drop!

**Please!** Make the desktop application for it (both Windows and Linux), so we can seamlessly drag and drop and manage the photos/videos from the computer ❤️
